### PR TITLE
chore: Upgrade relay to 10 

### LINKED ISSUE #3953: Questions about Relay
I have two questions to ask you:
  * why do you have react-relay to version 9.1.0 in the dependencies and its other dependencies to version 9.0.0?
  * why don't you upgrade the Relay version to 10.0.1?

Thanks

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Enhancement**

This resolves: **[MX-558]**

Fixes #3953 

### Description

This PR upgrades Relay to 10. See release notes: https://github.com/facebook/relay/releases/tag/v10.0.0. 

I've clicked around the app as much as I can and everything seems 👌. Upgrade process was smooth.   

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[MX-558]: https://artsyproduct.atlassian.net/browse/MX-558

### CODE REVIEW

File: src/lib/Scenes/ArtistSeries/__tests__/ArtistSeries-tests.tsx [hunk 1]
```diff
@@ -48,42 +48,42 @@ describe("Artist Series Rail", () => {
...
+  const getWrapper = (mockResolvers = {}) => {
     const tree = renderWithWrappers(<TestRenderer />)
     act(() => {
-      env.mock.resolveMostRecentOperation({
-        errors: [],
-        data: {
-          ...testFixture,
-        },
-      })
+      env.mock.resolveMostRecentOperation((operation) => MockPayloadGenerator.generate(operation, mockResolvers))
```
Comment (ashfurrow) [inline src/lib/Scenes/ArtistSeries/__tests__/ArtistSeries-tests.tsx:54]: Mmmm this is so much better 🤩

**Review by ashfurrow — APPROVED** [review]
Thank you again!

**Comment by ashfurrow** [comment]
I've added a link to a Jira ticket I'd opened to do this work after #3953 was opened. I'll resolve the conflicts and merge.

**Review by MounirDhahri — APPROVED** [review]

**Comment by ashfurrow** [comment]
I resolved a `patch-package` warning in c3ea5d8 but otherwise the merge went smoothly 👍 

#squashongreen

**Comment by damassi** [comment]
Thanks for the fix 👍

**Pull request merged**